### DISCUSSION TITLE
USB Encoder: button mapping and new functionallity

### DISCUSSION THREAD

**Discussion by sambold** [MiczFlor/RPi-Jukebox-RFID#2376] [Q&A] [answered]
hi, 
i connected a Dragonrise inc. generic usb joystick thingy as well as a few arcade buttons to my phoniebox and followed the installation instructions (https://github.com/MiczFlor/RPi-Jukebox-RFID/tree/develop/components/controls/buttons_usb_encoder). Everything works as intended and perfectly fine. 

my phoniebox has a few LEDs and if you push a button, different/random patterns are shown. now i want to map a key to execute this (user) function and maybe also configure some of the already mapped buttons/functions (like calling"playernext" by pressing the button and "playerseekfwd" by holding down the button).

if i understand the documentation correctly, i should be able to modify ~/RPi-Jukebox-RFID/settings/gpio_settings.ini to achieve what i want. but first of all changes in this file seem to have no effect to the button mapping and i also can't figure out which pin a specific button uses

i think this file might not be the right one for me, because as i use an usb encoder there are no pins which i can map to, right? but where can i find the mapping/settings-file for the usb encoder?

pretty sure this is kind of a stupid question and an easy to fix problem, but anyways i would really appreciate some help/input
thanks 
samuel

**Comment by AlvinSchiller** [comment, marked as answer]
Hi, 
the file `gpio_settings.ini` does only hold GPIO related mappings.
For the USB mappings a file `buttonMap.json` has been added at `components/controls/buttons_usb_encoder` from the setup.

Hope this helps.

**Reply by sambold** [reply]
thanks ... i was inspecting the github files and therefore i was missing that one.
